File: Data Structures & Algorithms/coin-change/submission-0.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        const int INF = amount + 1;

        vector<int> dp(amount + 1, INF);
        dp[0] = 0;

        for (int j = 1; j <= amount; ++j) {
            for (int coin : coins) {
                if (coin <= j) {
                    dp[j] = min(dp[j], 1 + dp[j - coin]);
                }
            }
        }

        return dp[amount] == INF ? -1 : dp[amount];
    }
};

File: Data Structures & Algorithms/stone-game/submission-0.py
class Solution:
    def stoneGame(self, piles: List[int]) -> bool:
        if len(piles) == 2:
            return True
        
        l, r = 0, len(piles) - 1
        a, b = 0, 0
        while l < r:
            a += max(piles[l], piles[r])
            b += min(piles[l], piles[r])
            l += 1
            r -= 1
        
        return True if a > b else False